libqtile/group.py

```diff
@@ -30,12 +30,90 @@
 
 import xcffib
 import xcffib.xproto
+from xcffib.xproto import StackMode
 
+from enum import Flag
 from libqtile import hook, utils, window
 from libqtile.command_object import CommandObject
 from libqtile.log_utils import logger
 
 
+class _History:
```

Comment:
Is it possible to add these smarts to ClientList somehow?

Reply:
That sounds very reasonable, I'll have to check the code of `_ClientList` in detail. Also, my initial intention was to clear the way for layering fake screens, which could make use of `_History` but not `_ClientList`. But that's for a future PR anyway, so maybe I'll not worry too much about it yet.

Reply:
The classes do similar things, but their scope seems entirely different: One is used by a group (and later, hopefully, by whatever manages screens) and allows only minimal interaction, the other is used by specific layouts and allows for much more control.

If anything, it looks like `_ClientList` could derive from `_History`. "current" in `_ClientList` would be translated to "top" in `_History`, I guess. But overall that would require an almost complete reimplementation of the class.

Reply:
As discussed with @ramnes, unifying the classes might be possible (although not trivial) and should be handled in a future PR.